.github/workflows/reproducible-build.yml

```
@@ -0,0 +1,169 @@
+
+          if contract_name:
+            logging.info(f"One contract will be built: {contract_name}.")
+
+          if create_release and attach_to_existing_release:
+            sys.exit("ERROR: Choose either to create a new release, or to attach the build output to an existing release. Can't pick both. Feel free to pick none, though.")
+          EOF
+        env:
+          image_tag: ${{ inputs.image_tag }}
+          contract_name: ${{ inputs.contract_name }}
+          create_release: ${{ inputs.create_release }}
+          attach_to_existing_release: ${{ inputs.attach_to_existing_release }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.GITHUB_REF_NAME }}
+          fetch-depth: 0
+          repository: ${{ env.GITHUB_REPOSITORY }}
+
+      - name: Preliminary checks
+        run: |
+          python3 - << "EOF"
+          import logging
+          from pathlib import Path
+          import sys
+
+          logging.basicConfig(level=logging.INFO)
+
+          logging.info("Check for missing 'wasm/Cargo.lock' files ...")
+
+          missing_cargo_lock = False
+
+          for cargo_toml in Path(".").rglob("wasm/Cargo.toml"):
+              cargo_lock = cargo_toml.with_suffix(".lock")
+              if cargo_lock.exists():
+                logging.info(f"Found wasm/Cargo.lock: {cargo_lock}")
+              else:
+                logging.error(f"wasm/Cargo.lock file not found: {cargo_lock}")
+                missing_cargo_lock = True
+            
+          if missing_cargo_lock:
+            sys.exit(f"ERROR: One or more 'wasm/Cargo.lock' files are missing. They are essential for reproducible builds.")
+          EOF
+
+      - name: Download build script
+        run: |
+          wget https://raw.githubusercontent.com/multiversx/mx-sdk-rust-contract-builder/${{ inputs.image_tag }}/build_with_docker.py
+      
+      - name: Build contracts
+        run: |
+          python3 ./build_with_docker.py --no-docker-tty --image=multiversx/sdk-rust-contract-builder:${{ inputs.image_tag }} --project=. --contract=${{ inputs.contract_name }} --output=/home/runner/work/output-from-docker
+      
+      - name: Save artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-output
+          path: |
+            /home/runner/work/output-from-docker/**/*.*
+          if-no-files-found: error
+
+      - name: Create new release (if applicable)
+        if: ${{ inputs.create_release == true }}
+        run: |
+          gh release create ${{ github.ref_name }} --target=$GITHUB_SHA --prerelease --title="New release: ${{ github.ref_name }}" --generate-notes
+          # Give some time for processing the request
+          sleep 10
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          image_tag: ${{ inputs.image_tag }}
+
+      - name: Create release notes
+        run: |
```

```suggestion
      - name: Create release notes
        if: ${{ inputs.attach_to_existing_release == true || inputs.create_release == true }}
        run: |
```
The release notes aren't saved as an artifact, so it would make sense to generate them only when working with a release.

---

Indeed, fixed.